salt/utils/versions.py

```diff
@@ -18,40 +15,63 @@
 import sys
 import warnings
 
-# pylint: disable=blacklisted-module
-try:
-    from setuptools.distutils.version import LooseVersion as _LooseVersion
-    from setuptools.distutils.version import StrictVersion as _StrictVersion
-except ImportError:
-    try:
-        from setuptools._distutils.version import LooseVersion as _LooseVersion
-        from setuptools._distutils.version import StrictVersion as _StrictVersion
-    except ImportError:
-        from distutils.version import LooseVersion as _LooseVersion
-        from distutils.version import StrictVersion as _StrictVersion
+import looseversion
+import packaging.version
 
-
-# pylint: enable=blacklisted-module
 import salt.version
 
-# pylint: disable=blacklisted-module
 log = logging.getLogger(__name__)
 
 
-class StrictVersion(_StrictVersion):
-    def parse(self, vstring):
-        _StrictVersion.parse(self, vstring)
+class Version(packaging.version.Version):
+    def __lt__(self, other):
+        if isinstance(other, str):
+            other = Version(other)
+        return super().__lt__(other)
 
-    def _cmp(self, other):
+    def __le__(self, other):
         if isinstance(other, str):
-            other = StrictVersion(other)
-        return _StrictVersion._cmp(self, other)
+            other = Version(other)
+        return super().__le__(other)
 
+    def __eq__(self, other):
+        if isinstance(other, str):
+            other = Version(other)
+        return super().__eq__(other)
 
-class LooseVersion(_LooseVersion):
-    def parse(self, vstring):
-        _LooseVersion.parse(self, vstring)
+    def __ge__(self, other):
+        if isinstance(other, str):
+            other = Version(other)
+        return super().__ge__(other)
 
+    def __gt__(self, other):
+        if isinstance(other, str):
+            other = Version(other)
+        return super().__gt__(other)
+
+    def __ne__(self, other):
+        if isinstance(other, str):
+            other = Version(other)
+        return super().__ne__(other)
+
+
+class StrictVersion(Version):
+    def __init__(self, *args, **kwargs):
```

Comment:
Wondering about this, marking it deprecated with the previous functionality removed, thus forcing change, since not backwards compatible with previous releases, since the previous implementation of StrictVersion differs from Version.  I am fine we doing this, but want to raise the issue that it is not real deprecation, since existing current functionality will have changed.